src/ale_interface.cpp

```diff
@@ -90,14 +90,12 @@ void ALEInterface::loadSettings(const string& romfile,
   // Seed random number generator
   if (theOSystem->settings().getString("random_seed") == "time") {
     cerr << "Random Seed: Time" << endl;
-    srand((unsigned)time(0));
-    //srand48((unsigned)time(0));
+    Random::seed((unsigned)time(0));
```

Comment:
Ok. I am going to change this.

On 24 May 2015 at 11:03, Marc G. Bellemare <notifications@github.com> wrote:

> In src/ale_interface.cpp
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55#discussion_r30953884>
> :
>
> > @@ -90,14 +90,12 @@ void ALEInterface::loadSettings(const string& romfile,
> >    // Seed random number generator
> >    if (theOSystem->settings().getString("random_seed") == "time") {
> >      cerr << "Random Seed: Time" << endl;
> > -    srand((unsigned)time(0));
> > -    //srand48((unsigned)time(0));
> > +    Random::seed((unsigned)time(0));
>
> time() takes in a pointer as argument. Please change to time(NULL). Also,
> can you cast to the correct unsigned type (uInt32)? The latter also applies
> elsewhere in this code.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mgbellemare/Arcade-Learning-Environment/pull/55/files#r30953884>
> .
>



-- 
Marlos Cholodovskis Machado
http://www.dcc.ufmg.br/~marlos <http://www.cs.ualberta.ca/~machado>